File: mRemoteV1/CS/Forms/OptionsPages/AdvancedPage.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.IO;
//using mRemoteNG.App.Info;
using mRemoteNG.My;
using mRemoteNG.Connection.Protocol;
using mRemoteNG.Tools;
using PSTaskDialog;


namespace mRemoteNG.Forms.OptionsPages
{
	public partial class AdvancedPage
		{
			public AdvancedPage()
			{
				InitializeComponent();
			}
#region Public Methods
public override string PageName
			{
				get
				{
					return Language.strTabAdvanced;
				}
				set
				{
				}
			}
			
			public override void ApplyLanguage()
			{
				base.ApplyLanguage();
				
				lblSeconds.Text = Language.strLabelSeconds;
				lblMaximumPuttyWaitTime.Text = Language.strLabelPuttyTimeout;
				chkAutomaticReconnect.Text = Language.strCheckboxAutomaticReconnect;
				lblConfigurePuttySessions.Text = Language.strLabelPuttySessionsConfig;
				btnLaunchPutty.Text = Language.strButtonLaunchPutty;
				btnBrowseCustomPuttyPath.Text = Language.strButtonBrowse;
				chkUseCustomPuttyPath.Text = Language.strCheckboxPuttyPath;
				chkAutomaticallyGetSessionInfo.Text = Language.strAutomaticallyGetSessionInfo;
				chkWriteLogFile.Text = Language.strWriteLogFile;
				lblUVNCSCPort.Text = Language.strUltraVNCSCListeningPort;
				lblXulRunnerPath.Text = Language.strXULrunnerPath;
				btnBrowseXulRunnerPath.Text = Language.strButtonBrowse;
				chkEncryptCompleteFile.Text = Language.strEncryptCompleteConnectionFile;
			}
			
			public override void LoadSettings()
			{
				base.SaveSettings();
				
				chkWriteLogFile.Checked = System.Convert.ToBoolean(My.Settings.Default.WriteLogFile);
				chkEncryptCompleteFile.Checked = System.Convert.ToBoolean(My.Settings.Default.EncryptCompleteConnectionsFile);
				chkAutomaticallyGetSessionInfo.Checked = System.Convert.ToBoolean(My.Settings.Default.AutomaticallyGetSessionInfo);
				chkAutomaticReconnect.Checked = System.Convert.ToBoolean(My.Settings.Default.ReconnectOnDisconnect);
				numPuttyWaitTime.Value = System.Convert.ToDecimal(My.Settings.Default.MaxPuttyWaitTime);
				
				chkUseCustomPuttyPath.Checked = MySettingsProperty.Settings.UseCustomPuttyPath;
				txtCustomPuttyPath.Text = MySettingsProperty.Settings.CustomPuttyPath;
				SetPuttyLaunchButtonEnabled();
				
				numUVNCSCPort.Value = System.Convert.ToDecimal(My.Settings.Default.UVNCSCPort);
				
				txtXULrunnerPath.Text = System.Convert.ToString(My.Settings.Default.XULRunnerPath);
			}
			
			public override void SaveSettings()
			{
				base.SaveSettings();
				
				My.Settings.Default.WriteLogFile = chkWriteLogFile.Checked;
				My.Settings.Default.EncryptCompleteConnectionsFile = chkEncryptCompleteFile.Checked;
				My.Settings.Default.AutomaticallyGetSessionInfo = chkAutomaticallyGetSessionInfo.Checked;
				My.Settings.Default.ReconnectOnDisconnect = chkAutomaticReconnect.Checked;
				
				bool puttyPathChanged = false;
				if (!(MySettingsProperty.Settings.CustomPuttyPath == txtCustomPuttyPath.Text))
				{
					puttyPathChanged = true;
					MySettingsProperty.Settings.CustomPuttyPath = txtCustomPuttyPath.Text;
				}
				if (!(MySettingsProperty.Settings.UseCustomPuttyPath == chkUseCustomPuttyPath.Checked))
				{
					puttyPathChanged = true;
					MySettingsProperty.Settings.UseCustomPuttyPath = chkUseCustomPuttyPath.Checked;
				}
				if (puttyPathChanged)
				{
					if (MySettingsProperty.Settings.UseCustomPuttyPath)
					{
						PuttyBase.PuttyPath = MySettingsProperty.Settings.CustomPuttyPath;
					}
					else
					{
						PuttyBase.PuttyPath = App.Info.General.PuttyPath;
					}
					Config.Putty.Sessions.AddSessionsToTree();
				}
				
				My.Settings.Default.MaxPuttyWaitTime = numPuttyWaitTime.Value;
				
				My.Settings.Default.UVNCSCPort = numUVNCSCPort.Value;
				
				My.Settings.Default.XULRunnerPath = txtXULrunnerPath.Text;
			}
#endregion
			
#region Private Methods
#region Event Handlers
			public void chkUseCustomPuttyPath_CheckedChanged(object sender, EventArgs e)
			{
				txtCustomPuttyPath.Enabled = chkUseCustomPuttyPath.Checked;
				btnBrowseCustomPuttyPath.Enabled = chkUseCustomPuttyPath.Checked;
				SetPuttyLaunchButtonEnabled();
			}
			
			public void txtCustomPuttyPath_TextChanged(object sender, EventArgs e)
			{
				SetPuttyLaunchButtonEnabled();
			}
			
			public void btnBrowseCustomPuttyPath_Click(object sender, EventArgs e)
			{
				using (OpenFileDialog openFileDialog = new OpenFileDialog())
				{
					openFileDialog.Filter = string.Format("{0}|*.exe|{1}|*.*", Language.strFilterApplication, Language.strFilterAll);
					openFileDialog.FileName = Path.GetFileName(App.Info.General.PuttyPath);
					openFileDialog.CheckFileExists = true;
					openFileDialog.Multiselect = false;
					
					if (openFileDialog.ShowDialog() == DialogResult.OK)
					{
						txtCustomPuttyPath.Text = openFileDialog.FileName;
						SetPuttyLaunchButtonEnabled();
					}
				}
				
			}
			
			public void btnLaunchPutty_Click(object sender, EventArgs e)
			{
				try
				{
					PuttyProcessController puttyProcess = new PuttyProcessController();
					string fileName = "";
					if (chkUseCustomPuttyPath.Checked)
					{
						fileName = txtCustomPuttyPath.Text;
					}
					else
					{
						fileName = App.Info.General.PuttyPath;
					}
					puttyProcess.Start(fileName);
					puttyProcess.SetControlText("Button", "&Cancel", "&Close");
					puttyProcess.SetControlVisible("Button", "&Open", false);
					puttyProcess.WaitForExit();
				}
				catch (Exception ex)
				{
					cTaskDialog.MessageBox(System.Convert.ToString(Application.Info.ProductName), Language.strErrorCouldNotLaunchPutty, "", ex.Message, "", "", eTaskDialogButtons.OK, eSysIcons.Error, null);
				}
			}
			
			public void btnBrowseXulRunnerPath_Click(object sender, EventArgs e)
			{
				FolderBrowserDialog oDlg = new FolderBrowserDialog();
				oDlg.ShowNewFolderButton = false;
				
				if (oDlg.ShowDialog() == DialogResult.OK)
				{
					txtXULrunnerPath.Text = oDlg.SelectedPath;
				}
				
				oDlg.Dispose();
			}
#endregion
			
			private void SetPuttyLaunchButtonEnabled()
			{
				string puttyPath = "";
				if (chkUseCustomPuttyPath.Checked)
				{
					puttyPath = txtCustomPuttyPath.Text;
				}
				else
				{
					puttyPath = App.Info.General.PuttyPath;
				}
				
				bool exists = false;
				try
				{
					exists = File.Exists(puttyPath);
				}
				catch
				{
				}
				
				if (exists)
				{
					lblConfigurePuttySessions.Enabled = true;
					btnLaunchPutty.Enabled = true;
				}
				else
				{
					lblConfigurePuttySessions.Enabled = false;
					btnLaunchPutty.Enabled = false;
				}
			}
#endregion
		}
}

File: mRemoteV1/Forms/OptionsPages/AdvancedPage.vb
Imports System.IO
Imports mRemote3G.App.Info
Imports mRemote3G.Config.Putty
Imports mRemote3G.Connection.Protocol
Imports mRemote3G.My
Imports mRemote3G.Tools
Imports PSTaskDialog

Namespace Forms.OptionsPages
    Public Class AdvancedPage

#Region "Public Methods"

        Public Overrides Property PageName As String
            Get
                Return Language.Language.strTabAdvanced
            End Get
            Set
            End Set
        End Property

        Public Overrides Sub ApplyLanguage()
            MyBase.ApplyLanguage()

            lblSeconds.Text = Language.Language.strLabelSeconds
            lblMaximumPuttyWaitTime.Text = Language.Language.strLabelPuttyTimeout
            chkAutomaticReconnect.Text = Language.Language.strCheckboxAutomaticReconnect
            lblConfigurePuttySessions.Text = Language.Language.strLabelPuttySessionsConfig
            btnLaunchPutty.Text = Language.Language.strButtonLaunchPutty
            btnBrowseCustomPuttyPath.Text = Language.Language.strButtonBrowse
            chkUseCustomPuttyPath.Text = Language.Language.strCheckboxPuttyPath
            chkAutomaticallyGetSessionInfo.Text = Language.Language.strAutomaticallyGetSessionInfo
            chkWriteLogFile.Text = Language.Language.strWriteLogFile
            lblUVNCSCPort.Text = Language.Language.strUltraVNCSCListeningPort
            chkEncryptCompleteFile.Text = Language.Language.strEncryptCompleteConnectionFile
        End Sub

        Public Overrides Sub LoadSettings()
            MyBase.SaveSettings()

            chkWriteLogFile.Checked = My.Settings.WriteLogFile
            chkEncryptCompleteFile.Checked = My.Settings.EncryptCompleteConnectionsFile
            chkAutomaticallyGetSessionInfo.Checked = My.Settings.AutomaticallyGetSessionInfo
            chkAutomaticReconnect.Checked = My.Settings.ReconnectOnDisconnect
            numPuttyWaitTime.Value = My.Settings.MaxPuttyWaitTime

            chkUseCustomPuttyPath.Checked = MySettingsProperty.Settings.UseCustomPuttyPath
            txtCustomPuttyPath.Text = MySettingsProperty.Settings.CustomPuttyPath
            SetPuttyLaunchButtonEnabled()

            numUVNCSCPort.Value = My.Settings.UVNCSCPort
        End Sub

        Public Overrides Sub SaveSettings()
            MyBase.SaveSettings()

            My.Settings.WriteLogFile = chkWriteLogFile.Checked
            My.Settings.EncryptCompleteConnectionsFile = chkEncryptCompleteFile.Checked
            My.Settings.AutomaticallyGetSessionInfo = chkAutomaticallyGetSessionInfo.Checked
            My.Settings.ReconnectOnDisconnect = chkAutomaticReconnect.Checked

            Dim puttyPathChanged = False
            If Not MySettingsProperty.Settings.CustomPuttyPath = txtCustomPuttyPath.Text Then
                puttyPathChanged = True
                MySettingsProperty.Settings.CustomPuttyPath = txtCustomPuttyPath.Text
            End If
            If Not MySettingsProperty.Settings.UseCustomPuttyPath = chkUseCustomPuttyPath.Checked Then
                puttyPathChanged = True
                MySettingsProperty.Settings.UseCustomPuttyPath = chkUseCustomPuttyPath.Checked
            End If
            If puttyPathChanged Then
                If MySettingsProperty.Settings.UseCustomPuttyPath Then
                    PuttyBase.PuttyPath = MySettingsProperty.Settings.CustomPuttyPath
                Else
                    PuttyBase.PuttyPath = General.PuttyPath
                End If
                Sessions.AddSessionsToTree()
            End If

            My.Settings.MaxPuttyWaitTime = numPuttyWaitTime.Value

            My.Settings.UVNCSCPort = numUVNCSCPort.Value
        End Sub

#End Region

#Region "Private Methods"

#Region "Event Handlers"

        Private Sub chkUseCustomPuttyPath_CheckedChanged(sender As Object, e As EventArgs) _
            Handles chkUseCustomPuttyPath.CheckedChanged
            txtCustomPuttyPath.Enabled = chkUseCustomPuttyPath.Checked
            btnBrowseCustomPuttyPath.Enabled = chkUseCustomPuttyPath.Checked
            SetPuttyLaunchButtonEnabled()
        End Sub

        Private Sub txtCustomPuttyPath_TextChanged(sender As Object, e As EventArgs) _
            Handles txtCustomPuttyPath.TextChanged
            SetPuttyLaunchButtonEnabled()
        End Sub

        Private Sub btnBrowseCustomPuttyPath_Click(sender As Object, e As EventArgs) _
            Handles btnBrowseCustomPuttyPath.Click
            Using openFileDialog As New OpenFileDialog()
                With openFileDialog
                    .Filter = String.Format("{0}|*.exe|{1}|*.*", Language.Language.strFilterApplication,
                                            Language.Language.strFilterAll)
                    .FileName = Path.GetFileName(General.PuttyPath)
                    .CheckFileExists = True
                    .Multiselect = False

                    If .ShowDialog = DialogResult.OK Then
                        txtCustomPuttyPath.Text = .FileName
                        SetPuttyLaunchButtonEnabled()
                    End If
                End With
            End Using
        End Sub

        Private Sub btnLaunchPutty_Click(sender As Object, e As EventArgs) Handles btnLaunchPutty.Click
            Try
                Dim puttyProcess As New PuttyProcessController
                Dim fileName As String
                If chkUseCustomPuttyPath.Checked Then
                    fileName = txtCustomPuttyPath.Text
                Else
                    fileName = General.PuttyPath
                End If
                puttyProcess.Start(fileName)
                puttyProcess.SetControlText("Button", "&Cancel", "&Close")
                puttyProcess.SetControlVisible("Button", "&Open", False)
                puttyProcess.WaitForExit()
            Catch ex As Exception
                cTaskDialog.MessageBox(Application.Info.ProductName, Language.Language.strErrorCouldNotLaunchPutty, "",
                                       ex.ToString(), "", "", eTaskDialogButtons.OK, eSysIcons.Error, Nothing)
            End Try
        End Sub

#End Region

        Private Sub SetPuttyLaunchButtonEnabled()
            Dim puttyPath As String
            If chkUseCustomPuttyPath.Checked Then
                puttyPath = txtCustomPuttyPath.Text
            Else
                puttyPath = General.PuttyPath
            End If

            Dim exists = False
            Try
                exists = File.Exists(puttyPath)
            Catch
            End Try

            If exists Then
                lblConfigurePuttySessions.Enabled = True
                btnLaunchPutty.Enabled = True
            Else
                lblConfigurePuttySessions.Enabled = False
                btnLaunchPutty.Enabled = False
            End If
        End Sub

#End Region
    End Class
End Namespace